test/e2e/clusterctl_upgrade_test.go

```diff
@@ -23,14 +23,68 @@ import (
 	. "github.com/onsi/ginkgo/v2"
 )
 
-var _ = Describe("When testing clusterctl upgrades [clusterctl-Upgrade]", func() {
+var _ = Describe("When testing clusterctl upgrades (v0.3=>current)", func() {
 	ClusterctlUpgradeSpec(ctx, func() ClusterctlUpgradeSpecInput {
 		return ClusterctlUpgradeSpecInput{
-			E2EConfig:             e2eConfig,
-			ClusterctlConfigPath:  clusterctlConfigPath,
-			BootstrapClusterProxy: bootstrapClusterProxy,
-			ArtifactFolder:        artifactFolder,
-			SkipCleanup:           skipCleanup,
+			E2EConfig:                 e2eConfig,
+			ClusterctlConfigPath:      clusterctlConfigPath,
+			BootstrapClusterProxy:     bootstrapClusterProxy,
+			ArtifactFolder:            artifactFolder,
+			SkipCleanup:               skipCleanup,
+			InitWithBinary:            "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.25/clusterctl-{OS}-{ARCH}",
```

Comment:
I'm debated about having something versioned here (binary, kubernetes version), this adds one more point where we have to keep them up to date 🤔

Reply:
What we can do is also consume this from the e2e conf / docker.yaml, e.g.:
```
INIT_WITH_BINARY_V1ALPHA3
INIT_WITH_PROVIDERS_CONTRACT_V1ALPHA3
INIT_WITH_KUBERNETES_VERSION_V1ALPHA3

INIT_WITH_BINARY_V1ALPHA4
INIT_WITH_PROVIDERS_CONTRACT_V1ALPHA4
INIT_WITH_KUBERNETES_VERSION_V1ALPHA4

INIT_WITH_BINARY_V1BETA1
INIT_WITH_PROVIDERS_CONTRACT_V1BETA1
INIT_WITH_KUBERNETES_VERSION_V1BETA1
```
This way:
* we still have it here in our repo and don't have to maintain external jobs
* we can modify them when modifying similar stuff in docker.yaml
* we can overwrite them if we want to run those jobs locally with different versions
* those env vars would be specific to our CAPD tests (I would just read that from the e2eConfig in _test.go)

WDYT?

Reply:
after some thinking, we can go with the proposed change, and rework if we find this complex to maintain down the line